python/tvm/topi/sort.py

```diff
@@ -161,3 +203,25 @@ def topk(data, k=1, axis=-1, ret_type="both", is_ascend=False, dtype="int64"):
         tag="topk_cpu",
     )
     return out
+
+
+@tvm.target.generic_func
+def dyn_topk_legalize(attrs, inputs, types):
+    """Legalizes dyn.topk op.
```

Comment:
Can you add a brief description of why legalization is needed here and what transformation is applied?